lib/types/json.js

```diff
@@ -77,10 +77,13 @@ function json(options) {
 
     if (strict) {
       var first = firstchar(body)
-
+      var message = 'Unexpected token ' + first
       if (first !== '{' && first !== '[') {
+        if (process.version[1] >= 6) {
+          message += ' in JSON at position 1'
```

Comment:
dawning on me that I misunderstood what firschar did...

could we replace the static number with `body.search(first)`?